src/overlays/actors/ovl_En_Fr/z_en_fr.c

```diff
@@ -8,14 +9,169 @@ void EnFr_Init(Actor* thisx, GlobalContext* globalCtx);
+void EnFr_SetIdle(EnFr* this, GlobalContext* globalCtx);
+
+// Drawing Functions
+void EnFr_DrawIdle(EnFr* this);
+void EnFr_DrawActive(EnFr* this);
+s32 EnFr_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, void* thisx,
+                          Gfx** gfx);
+void EnFr_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, void* this);
+void EnFr_Draw(Actor* thisx, GlobalContext* globalCtx);
+
+extern FlexSkeletonHeader D_0600B498; // Frog
+extern AnimationHeader D_06001534;    // Frog
+extern FlexSkeletonHeader D_050036F0; // Butterfly
+extern AnimationHeader D_05002470;    // Butterfly
+extern AnimationHeader D_060007BC;    // Frog Jumping
+extern AnimationHeader D_060011C0;    // Frog Landing
+
+//
+// Frogs actor.params system
+//
+// Represents 6 Actor Instances for frogs:
+//     - 1 Prop actor instance set to where Link plays Ocarina, manages 5 frogs
+//     - 5 NPC actor instances for the frogs themselves
+//
+// 0: Prop Actor Instance (located where link detects ocarina, interacts with Link)
+// 1: Frog 0 (Yellow)
+// 2: Frog 1 (Blue)
+// 3: Frog 2 (Red)
+// 4: Frog 3 (Purple)
+// 5: Frog 4 (White)
+//
+// Note that because of the Prop Actor, actor.params is 1 shifted from frogIndex
+// Therefore, frogIndex = actor.params - 1
+
+//
+// sEnFrPointers.flags system
+//
+//
+// sEnFrPointers.flags = 0x01
+//      - Activate frogs, frogs will jump out of the water
+//
+// sEnFrPointers.flags = 0x01 to 0x0B:
+//      - Counter: Frogs will sequentially jump out based on counter:
+//          - 0x01: Frog 1 (Blue)
+//          - 0x03: frog 3 (Purple)
+//          - 0x05: frog 0 (Yellow)
+//          - 0x07: frog 2 (Red)
+//          - 0x09: frog 4 (White)
+//      - Will proceed when counter reachers 0x0B
+//
+// sEnFrPointers.flags = 0x0C
+//      - Deactivate frogs, frogs will jump back into the water
+//
+
+typedef struct {
+    u8 flags;
+    EnFr* frogs[5]; // Frog Pointers
+} EnFrPointers;
+
+typedef struct {
+    f32 xzDist;
+    f32 yaw;
+    f32 yDist;
+} LogSpotToFromWater;
+
+typedef struct {
+    u8 red;
+    u8 green;
+    u8 blue;
+    u8 alpha;
+} EnFrColor;
+
+// A pointer to each NPC froc actor instance on the heap
+EnFrPointers sEnFrPointers = {
+    0x00,
+    { NULL,  // FROG_YELLOW
+      NULL,  // FROG_BLUE
+      NULL,  // FROG_RED
+      NULL,  // FROG_PURPLE
+      NULL } // FROG_WHITE
+};
+
+// Flags for gSaveContext.eventChkInf[13]
+u16 sSongIndex[] = { 0x0002,   // ZL            (second bit)
+                     0x0004,   // Epona         (third bit)
+                     0x0010,   // Saria         (fifth bit)
+                     0x0008,   // Suns          (fourth bit)
+                     0x0020,   // SoT           (sixth bit)
+                     0x0040,   // Storms        (seventh bit)
+                     0x0001,   // Frogs 2 Song  (first bit)
+                     0x0000 }; // Padding
```

Comment:
for multiline lists put a trailing comma on the last entry so the formatter doesnt do this weird indenting.
I also think the bit comments are a bit redundant, theyll prob become macros later anyway

Reply:
The bit comments are a bit excessive yeah. A lot of extra comments were really for my own benefit just to help with understanding the code lol. For the songs, would it be easy to get which bits represents which songs for mappings in general? I know I spent time on gz figuring out which bits/index/flags represents which songs/controller notes in general. Should I leave the comments in to help name the macros when that time comes? Or is that fairly easy without the comments? I have no preference either way.

Reply:
whoever gets to audio stuff will take care of it, but you can leave the song comments for the time being to make it easier to reference until that point. i was talking specifically about `(first bit)` etc

Reply:
also, the padding should be able to be omitted. the compiler should insert it on its own

Reply:
I'm also tempted to make an enum for the songs (mainly for sFrogToSongIndex). That might also work and be more readable, but feel free to object

Reply:
should be fine. they may end up getting replaced by more general audio defines down the line, but since we dont know what those will be yet, we can figure it out later